File: tests/fixtures/src/sample.py
"""Sample Python module for golden snapshot testing."""
from __future__ import annotations

import os
from typing import Optional


class DataProcessor:
    """Processes raw data records."""

    def __init__(self, path: str) -> None:
        self.path = path
        self._cache: dict[str, list] = {}

    def load(self, key: str) -> Optional[list]:
        """Load records for the given key."""
        if key in self._cache:
            return self._cache[key]
        return None

    def process(self, records: list) -> dict:
        """Transform records into a summary."""
        result: dict = {}
        for item in records:
            result[item["id"]] = item
        return result


def format_output(data: dict, indent: int = 2) -> str:
    """Render data as a formatted string."""
    import json
    return json.dumps(data, indent=indent)

File: tests/fixtures/src/sample.ts
// Sample TypeScript module for golden snapshot testing.
import { readFileSync } from "fs";

export interface Record {
  id: string;
  value: number;
}

export type ProcessFn = (r: Record) => string;

export class DataService {
  private records: Record[] = [];

  constructor(private readonly path: string) {}

  load(): void {
    const raw = readFileSync(this.path, "utf-8");
    this.records = JSON.parse(raw);
  }

  process(): string[] {
    return this.records.map((r) => `${r.id}:${r.value}`);
  }
}

export const formatRecord: ProcessFn = (r) => `${r.id}=${r.value}`;
